### ISSUE TITLE
Capture pen button input to to delete strokes on Xournal++

### ISSUE THREAD

**Issue by shubhampaul** [H-M-H/Weylus#66]
As the heading say, is there a way to capture button input of my S-pen (on Galaxy Tab S3) and then feed it to Xournal++ where it can be configured as an eraser?

Xournal++ easily captures button input of pens from Pen Tabs, I was wondering if it would be possible with my S-pen as well?

**Comment by H-M-H** [comment]
This is currently not implemented as I don't have a pen with button I could test it with. But I think it is fairly straightforward to implement. If you want to help, please run the latest version of Weylus like this: `WEYLUS_LOG_LEVEL=TRACE ./weylus` and paste the output you get after using the pen and pressing the button here. With some luck the browser sends a proper button event I can use.

**Comment by shubhampaul** [comment]
@H-M-H Sorry for the delay. Here is the log from Weylus app on Linux, please let me know if that's what you wanted me to post. [Debug Log](https://pastebin.com/LZTfpN8c ). An observation- when I press the button and try to scribble, the cursor doesn't move at all.

Note: I tried both by enabling/disabling the option "Produce touch events".

**Comment by H-M-H** [comment]
Yes, this is the log I want but please try again with the latest version :)

**Comment by shubhampaul** [comment]
Here is the log from the latest version (0.10.0) [Debug Log 3](https://pastebin.com/tyPBzzKC).

**Comment by H-M-H** [comment]
Alright, please try the latest build: https://github.com/H-M-H/Weylus/actions/runs/647438053

**Comment by shubhampaul** [comment]
Here is the log.
[Debug Log 4.txt](https://github.com/H-M-H/Weylus/files/6134931/Debug.Log.4.txt)

**Comment by H-M-H** [comment]
Hmm, I am afraid but this looks like bad news. I had a look at the specs for `PointerEvents` again and found a button code specifically for the eraser function of a pen. So I implemented it but it looks like this button code is never generated on your tablet.

Please try the (new) latest build: https://github.com/H-M-H/Weylus/actions/runs/649368278 it will print a warning in case an unknown button code is generated (I previously ignored these and there is a slim chance that your eraser might generate one).

**Comment by shubhampaul** [comment]
Here is the new log 
[Debug Log 5.txt](https://github.com/H-M-H/Weylus/files/6136647/Debug.Log.5.txt)

I am using chrome on my tablet, I try with firefox once and upload the log.

I was wondering about this approach (please let me know if it is possible or not): Since the button is detected in hovering mode (pressure: 0.0), can you simply add a condition to send two different inputs to the PC (Xournal++)? Like detect the event as eraser button input when 'buttons: NONE' changes to 'buttons: PRIMARY' **and** 'pressure **=** 0.0' and the second event can detect a 'normal pen' when the when 'buttons: NONE' changes to 'buttons: PRIMARY' **and** 'pressure **>** 0.0' Please don't quote on my use of the words 'event/input' but I think you understand the gist of what I am trying to say. 

This way the S-pen can be used as a hovering eraser when pressing and holding the button but not touching the screen and a normal pen when touching the screen and not holding the button. 

I mean this could be a simple way but to get close to a 'more true' pen eraser style, one could add more sophisticated conditional 'if's so that it erases only when touching the screen (as long as the button is being pressed but with the condition that one has to press and hold the button before making the pen touch the screen as it works with my 'Samsung Notes' app). Before this approach, there need to be some changes that need to be made in your code I suppose since I have noticed that I stop getting log output when I press the button and touch the screen. 1) Simple touching gives log, 2) Simple hovering gives log, 3) Hovering while pressing the button gives log but 4) Touching the screen while pressing the button doesn't give a log (and the pointer also doesn't move). So 3) and/or 4) can be used for eraser mode. 

Please let me know what you think of it.

**Comment by H-M-H** [comment]
Reading through the source of chrome it becomes evident that pressing the button of a pen is intercepted and never reaches JavaScript-land. Instead pressing it will start text selection: Interception happens [here](https://source.chromium.org/chromium/chromium/src/+/master:content/browser/renderer_host/render_widget_host_view_android.cc;l=946-949;drc=aace0cfef2d51e7f038b07d5df26d126e4160f70). Detection of button press happens [here](https://source.chromium.org/chromium/chromium/src/+/master:content/browser/renderer_host/input/stylus_text_selector.cc;l=144;drc=40f5195e60a82e940f4ef149e55944da9c8bf1f0). Notable is that text selection only starts if the pen touches the screen / is in the `down` state. This also explains why you could observe button presses while hovering the pen.

This means it's not possible to get the eraser working using the button of the pen if the pen also touches the screen (at least with chrome). It is possible to simulate eraser events as you proposed if the pen is only hovering, I will think about adding that.

**Comment by H-M-H** [comment]
And looking at firefox it seems that `pen` events are not implemented for android: https://searchfox.org/mozilla-central/search?q=symbol:_ZN7mozilla3dom18MouseEvent_BindingL14MOZ_SOURCE_PENE&redirect=false

**Comment by H-M-H** [comment]
I have opened an issue for Firefox here: https://github.com/mozilla-mobile/fenix/issues/18749

**Comment by NicoWeio** [comment]
If you need another tester in the future, I have a Samsung Galaxy Tab S3 as well. :)

**Comment by tabbas97** [comment]
Hi @H-M-H 
Probably not the best place to introduce myself but I think I can help with testing too. I have a Samsung Tab S7 FE and use Ubuntu and Windows interchangeably. 

This is one of the most useful tools I have used in the past year
